compiler/src/codegen/compcore.re

```diff
@@ -7,6 +7,13 @@ open Grain_utils;
 open Comp_utils;
 open Comp_wasm_prim;
 
+/*
+ GENERAL NOTES (extend as desired)
+ =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+ Make sure that all nodes in the Binaryen AST are distinct instances! See note in link.re.
+ */
```

Comment:
I kind of have a lot of thoughts and feelings about this, so as not to block this change can we pull this and the note out of this PR? I can get a PR up with some info around this.

Reply:
#743